### PR TITLE
Remove UT data structures from global scheduler.

### PR DESCRIPTION
**Pull request by robertnishihara**
Note this also removes a bunch of unused code that I thought would be easier to rewrite later than to convert.

More progress on #404.

### CODE REVIEW

**Comment by robertnishihara** [comment]
test this please

File: src/global_scheduler/global_scheduler.cc [hunk 1]
```diff
@@ -61,7 +61,6 @@ GlobalSchedulerState *GlobalSchedulerState_init(event_loop *loop,
                                                 int redis_primary_port) {
   GlobalSchedulerState *state = new GlobalSchedulerState();
   /* Must initialize state to 0. Sets hashmap head(s) to NULL. */
```
Comment (pcmoritz) [inline src/global_scheduler/global_scheduler.cc]: Get rid of this comment too

File: src/global_scheduler/global_scheduler.cc [hunk 2]
```diff
@@ -332,9 +329,9 @@ int heartbeat_timeout_handler(event_loop *loop, timer_id id, void *context) {
        * next iterator. */
       it = remove_local_scheduler(state, it);
     } else {
+      ++it->second.num_heartbeats_missed;
```
Comment (pcmoritz) [inline src/global_scheduler/global_scheduler.cc]: 'it->second.num_heartbeats_missed += 1' is clearer here

File: src/global_scheduler/global_scheduler_algorithm.cc [hunk 3]
```diff
@@ -268,9 +78,70 @@ double calculate_score_dynvec_normalized(GlobalSchedulerState *state,
   return score;
 }
 
+int64_t calculate_data_per_local_scheduler(const GlobalSchedulerState *state,
```
Comment (pcmoritz) [inline src/global_scheduler/global_scheduler_algorithm.cc]: call this already_available_data_size

**Pull request merged**